### PR TITLE
Add RelayPlugin

### PR DESCRIPTION
**Pull request by kassens**
This is mostly @jaredly's work to add an initial plugin for Relay.

It currently includes #186, but I'll rebase once that's landed to clean this PR up.

I've fixed lint and flow issues as well as actually disabled this plugin by default. Since we don't have options yet, it's just commented out in `Panel.js`.

It's the beginning of #187.

### CODE REVIEW

**Comment by kassens** [comment]
@yungsters @steveluscher @spicyj Thoughts on moving forward with this?

File: plugins/Relay/Store.js [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+      dataIDs.forEach(id => {
+        if (!this.queriesByDataID[id]) {
+          this.queriesByDataID[id] = [data.id];
+        } else {
+          this.queriesByDataID[id].push(data.id);
+        }
+      });
+    });
+    bridge.on('relay:success', ({id, response, end}) => {
+      console.log('response', id, response);
```
Comment (yungsters) [inline plugins/Relay/Store.js]: We should remove these.

**Comment by kassens** [comment]
@yungsters and I thought it's best to land this (behind the URL toggle) and then iterate.

**Comment by jaredly** [comment]
At one point I had a settings tab [see branch here](https://github.com/jaredly/react-devtools/tree/settings-tab), where I planned to allow users to enable/disable various plugins. That way you wouldn't need the location.hash thing.

**Comment by kassens** [comment]
@jaredly: Yeah, I saw that but it didn't seem worth it for a single option. Once, we evolve the plugin a bit more, I think we can enable most things just when we see it's being used on the page.

**Comment by jaredly** [comment]
Other things that a settings page would be useful for:
Switching off display of `<RelayWrapper/>` components, see #211 

I agree that relay should only appear when it's detected though

**Pull request merged**